hedera-node/cli-clients/src/main/java/com/hedera/services/cli/signedstate/DumpTopicsSubcommand.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.services.cli.signedstate;
+
+import static com.hedera.services.cli.utils.ThingsToStrings.getMaybeStringifyByteString;
+import static com.hedera.services.cli.utils.ThingsToStrings.quoteForCsv;
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.node.app.service.mono.legacy.core.jproto.JKey;
+import com.hedera.node.app.service.mono.state.adapters.MerkleMapLike;
+import com.hedera.node.app.service.mono.state.merkle.MerkleTopic;
+import com.hedera.node.app.service.mono.state.submerkle.EntityId;
+import com.hedera.node.app.service.mono.state.submerkle.RichInstant;
+import com.hedera.node.app.service.mono.utils.EntityNum;
+import com.hedera.services.cli.signedstate.DumpStateCommand.EmitSummary;
+import com.hedera.services.cli.signedstate.DumpTokensSubcommand.FieldBuilder;
+import com.hedera.services.cli.signedstate.SignedStateCommand.Verbosity;
+import com.hedera.services.cli.utils.ThingsToStrings;
+import com.hedera.services.cli.utils.Writer;
+import com.swirlds.base.utility.Pair;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.TreeMap;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/** Dump all topics from a signed state file to a text file in a deterministic order  */
+public class DumpTopicsSubcommand {
+
+    static void doit(
+            @NonNull final SignedStateHolder state,
+            @NonNull final Path topicsPath,
+            @NonNull final EmitSummary emitSummary,
+            @NonNull final Verbosity verbosity) {
```

Comment:
Do we need the `verbosity` field? Looks like it's not currently used.

Reply:
I can relate, but it looks like the same approach where verbosity isn't used is also applied in [DumpTokensSubcommand](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/cli-clients/src/main/java/com/hedera/services/cli/signedstate/DumpTokensSubcommand.java#L70), [DumpTokenAssociationsSubcommand](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/cli-clients/src/main/java/com/hedera/services/cli/signedstate/DumpTokenAssociationsSubcommand.java), [DumpUniqueTokensSubcommand](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/cli-clients/src/main/java/com/hedera/services/cli/signedstate/DumpUniqueTokensSubcommand.java#L54).

@david-bakin-sl  What do you think should we leave it for future uses?

Reply:
yes, I'd leave it - I like the option for all the dumpers to have the option to dump stuff verbosely - even if any given dumper doesn't do it.  I think the uniformity is easier to read, if you're reading more than one of them.  IMO.